src/config.rs

```diff
@@ -24,6 +26,17 @@ pub struct Config {
     /// The policy to use when warnings are encountered.
     #[serde(default)]
     pub warning_policy: WarningPolicy,
+    /// The map of regexes representing sets of web sites and
+    /// the list of HTTP headers that must be sent to matching sites.
+    #[serde(with = "headers_serde")]
+    pub http_headers: Vec<(Regex, Vec<HttpHeader>)>,
+}
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Clone)]
+#[serde(try_from = "String", into = "String")]
+pub struct HttpHeader {
```

Comment:
Does the `http` crate expose some sort of header key-value type? If so, we may want to reuse that instead of rolling our own.

Reply:
It exposes separate `HeaderName` and `HeaderValue` types, the split is up to us anyway.